src/js/index.js

```diff
@@ -0,0 +1,138 @@
+import styles from '../styles/app.module'
+import '../styles/global'
+import plpView from './views/plpView'
+import cart from './models/cart'
+import cartView from './views/cartView'
+var bannerTemplate = require("./templates/banners.hbs");
+var categoriesBannerTemplate = require("./templates/categoriesBanner.hbs");
+
+var cartViewObj = new cartView();
+var cartObj = new cart(cartViewObj);
+
+var controller = (function() {
+  
+      const plpPage = new plpView();
+
+
+      function initCart(){
+        console.log("Controller initialised"); 
+          
+      }
+
+      function setBuyBtnEventListner(){
+        var btns = document.querySelectorAll(".bn-btn, .sm-bn-btn, .xsm-bn-btn");
+        
+        btns.forEach(btn => {
+            btn.onclick = function(){
+            cartObj.addItem(btn.parentElement.parentElement.id || btn.parentElement.parentElement.parentElement.id);
+            
+            
+          }
+        });
+      }
+
+      function setCategoryListeners(){
+        var categories = document.querySelectorAll(".category");
+        categories.forEach(category => {
+          category.onclick = function(event){
+            var categoryId= event.target.id || (event.target.id).split("-")[1];
+            plpPage.RenderFilteredItems(categoryId);
+            setBuyBtnEventListner();
+        }
+      });
+
+      }
+
+      function loadBanners(){
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.open( "GET", "api/banners/index.get.json", false ); // false for synchronous request
```

Comment:
You should never make synchronous AJAX request; it defeats the purpose all togther